pkg/controller/workspace/workspace_controller.go

```diff
@@ -228,18 +236,21 @@ func (r *ReconcileWorkspace) Reconcile(request reconcile.Request) (reconcileResu
 		if componentsStatus.FailStartup {
 			reqLogger.Info("Workspace start failed")
 			reconcileStatus.Phase = devworkspace.WorkspaceStatusFailed
+			// TODO: Propagate more information from sync step to show a more useful message -- which plugin couldn't be installed?
+			reconcileStatus.Conditions[WorkspaceFailedStart] = "Could not find plugins for devworkspace"
```

Comment:
When it's going to be shown? Currently, I see starting workspace without any condition when I configure non-existing plugin version.

Reply:
This is a limitation that should be resolved via propagating messages better -- currently, this error will be set when
1. An unsupported component type is in the devfile (e.g. "dcokerfile")
2. The default dockerimage component is not set and the devfile does not contain one
3. We fail to set ownerref on the components subresource (which should *not* be a failure at all)

The error message doesn't match any of these cases, but there's also not really one message that can explain the issue. I've created https://github.com/eclipse/che/issues/17455 to resolve this (it should address the three points above and also propagate errors from the subcontroller in the same way we'll approach workspaceRouting).